tests/runfiles/linux.run

```diff
@@ -348,6 +348,9 @@ tests = ['zpool_upgrade_001_pos',
     'zpool_upgrade_006_neg', 'zpool_upgrade_008_pos',
     'zpool_upgrade_009_neg']
 
+[tests/functional/cli_root/zpool_sync]
+tests = ['zpool_sync_001_pos', 'zpool_sync_002_neg']
```

Comment:
Test cases appear to be missing from the PR.

We can also vastly improve the test coverage by updating the `sync_pool` helper function in `libtest.shlib` to simply call `zpool sync`.  I'd also suggest removing the other versions of `sync_pool` in `clean_mirror_common.kshlib`, and `txg_sync` in `create-o_ashift.ksh`.